src/main/java/com/lambdatest/tunnel/Tunnel.java

```diff
@@ -584,58 +553,56 @@ public void runCommand(String command) throws IOException {
 
     public void runCommandV2(String[] command) throws IOException {
         try {
-            // ProcessBuilder processBuilder = new ProcessBuilder(command);
             Runtime run = Runtime.getRuntime();
             process = run.exec(command);
             Boolean update = false;
-            long start = System.currentTimeMillis();
-            long end = start;
+            waitForTunnelToStart();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void waitForTunnelToStart() throws IOException {
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(HTTP_TIMEOUT)
+                .setSocketTimeout(HTTP_TIMEOUT)
+                .setConnectionRequestTimeout(HTTP_TIMEOUT)
+                .build();
+        CloseableHttpClient httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build();

```

Comment:
httpClient not closed, can we use a try-with-resource statement here something like -

 try (CloseableHttpClient httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build()) {
// rest of logic here
 }

Reply:
done